### PR TITLE
Issue #12878: Resolve pitest suppression for BeforeExecutionFileFilterSet

### PR DESCRIPTION
**Pull request by shubh220922**
Related to #12878

@romani 
Surviving mutation has been killed.
Pure JUnit testing has been used here as well.
Reason for which has been stated in a multiline comment .

### CODE REVIEW

**Comment by nrmancuso** [comment]
@shubh220922 please rebase and resolve conflicts

File: src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java [hunk 1]
```diff
@@ -115,4 +116,10 @@ public void testUnmodifiableSet() {
             () -> filterSet.getBeforeExecutionFileFilters().add(filter));
     }
 
+    @Test
```
Comment (nrmancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java:119]: Please make a comment on method with details on why we kill this mutation with pure unit test.

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java (hunk 1 above)
Comment (shubh220922) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java:119]: ok. Missed that while fixing an error probably.

Comment (shubh220922) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java:123]: @nrmancuso 
Also jupiter assertion api have been used here. Is that fine?

File: src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java [hunk 2]
```diff
@@ -115,4 +116,14 @@ public void testUnmodifiableSet() {
...
     }
 
+    /*
+      This comment is not related to this code. Input based test does not call
+      toString, but this method might be useful for third party integrations
+    */
+    @Test
+    public void testEmptyToString() {
+        final BeforeExecutionFileFilterSet filterSet = new BeforeExecutionFileFilterSet();
+        Assertions.assertNotEquals(filterSet.toString(), "");
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java:126]: Truth

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java [hunk 3]
```diff
@@ -115,4 +116,14 @@ public void testUnmodifiableSet() {
             () -> filterSet.getBeforeExecutionFileFilters().add(filter));
     }
 
+    /*
+      This comment is not related to this code. Input based test does not call
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java:120]: `This comment is not related to this code.`

Remove this

**Review by romani — CHANGES_REQUESTED** [review]

File: src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java (hunk 3 above)
Comment (shubh220922) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java]: @romani Done. 
i was getting PMD violations for this PR locally. Turns out PMD wouldn't allow whatever test I had written previously. 
image
I have fixed it though.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java (hunk 2 above)
Comment (shubh220922) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java]: @romani Done.

Comment (shubh220922) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java]: @romani Done

File: src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java (hunk 3 above)
Comment (shubh220922) [reply src/test/java/com/puppycrawl/tools/checkstyle/api/BeforeExecutionFileFilterSetTest.java]: @romani Done.

**Comment by shubh220922** [comment]
@romani please review.

**Review by romani — APPROVED** [review]
ok to merge
